pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/codestyle/xml/MethodNamingConventions.xml

```diff
@@ -81,10 +70,11 @@ public class Foo {
     <test-code>
         <description>#1573 method names should not contain underscores, but skip test methods 2</description>
         <rule-property name="skipTestMethodUnderscores">true</rule-property>
-        <expected-problems>0</expected-problems>
+        <expected-problems>1</expected-problems>
         <code><![CDATA[
 public class Foo {
-    void testMethod test_barFoo() {}
```

Comment:
without a testmethod, this test doesn't test anymore, what it should... 

Interesting (https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation_isTest.htm):

> The testMethod keyword is now deprecated. Use the @ isTest annotation on classes and methods instead. The 
> @ isTest annotation on methods is equivalent to the testMethod keyword. 

maybe the syntax is this:

```
public class Foo {
    testmethod void test_barFoo() {}
}
```

It could also be, that the testMethod keyword already produces a parsing error, since it's deprecated.

The new way should look like this:

```
public class Foo {
    @isTest void test_barFoo() {}
}
```

If possible, the test case should test both variants....

Reply:
Yes, with the keyword, it works again. We have already a test using the annotation, so no new test is necessary.

Reply:
Hmm interesting, I didn't find doc about this keyword, I just thought it was a bad copy paste